### ISSUE TITLE
jdk16 aarch64 crash running cmdLineTester_decompilationTests_0

### ISSUE THREAD

**Issue by lumpfish** [eclipse/openj9#12217]
cmdLineTester_decompilationTests_0 failed running on the jdk-16_openj9-0.25.0 build:
```
openjdk version "16-ea" 2021-03-16
OpenJDK Runtime Environment AdoptOpenJDK (build 16-ea+36)
Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.25.0, JRE 16 Linux aarch64-64-Bit Compressed References 20210311_79 (JIT enabled, AOT enabled)
OpenJ9   - 022d65424
OMR      - 09514431e
JCL      - 0c11227a21 based on jdk-16+36)
```
cmdLineTester_decompilationTests_1 (non-compressed refs) also failed in the same way.

Link to failing job: https://ci.adoptopenjdk.net/job/Test_openjdk16_j9_sanity.functional_aarch64_linux/46/

Output from failing test:
```
13:58:17  ===============================================
13:58:17  Running test cmdLineTester_decompilationTests_0 ...
13:58:17  ===============================================
13:58:17  cmdLineTester_decompilationTests_0 Start Time: Thu Mar 11 13:58:12 2021 Epoch Time (ms): 1615471092387
13:58:17  "/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux/openjdkbinary/j2sdk-image/bin/java" -Xshareclasses:destroyAll; "/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux/openjdkbinary/j2sdk-image/bin/java" -Xshareclasses:groupAccess,destroyAll; echo "cache cleanup done";
13:58:17  JVMSHRC005I No shared class caches available
13:58:17  JVMSHRC005I No shared class caches available
13:58:17  cache cleanup done
13:58:17  variation: Mode112-OSR
13:58:17  JVM_OPTIONS:  -Xgcpolicy:gencon -Xjit:enableOSR,count=0 
13:58:17  { itercnt=1; \
13:58:17  mkdir -p "/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux/openjdk-tests/TKG/../TKG/output_16154637976441/cmdLineTester_decompilationTests_0"; \
13:58:17  cd "/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux/openjdk-tests/TKG/../TKG/output_16154637976441/cmdLineTester_decompilationTests_0"; \
13:58:17  export LD_LIBRARY_PATH="/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux/openjdkbinary/openjdk-test-image/openj9:/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux/openjdkbinary/j2sdk-image/bin/../lib/default:/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux/openjdkbinary/j2sdk-image/bin/../lib/j9vm:"; \
13:58:17  "/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux/openjdkbinary/j2sdk-image/bin/java"  -Xgcpolicy:gencon -Xjit:enableOSR,count=0  \
13:58:17  -DTEST_ROOT"/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux/openjdk-tests/TKG/../../jvmtest/functional/cmdLineTests/jvmtitests" \
13:58:17  -DJAR="/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux/openjdk-tests/TKG/../../jvmtest/functional/cmdLineTests/jvmtitests/jvmtitest.jar" \
13:58:17  -DEXE='"/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux/openjdkbinary/j2sdk-image/bin/java"  -Xgcpolicy:gencon -Xjit:enableOSR,count=0  -Xdump -Xgcthreads1 ' \
13:58:17  -jar "/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux/openjdk-tests/TKG/../../jvmtest/functional/cmdline_options_tester/cmdlinetester.jar" \
13:58:17  -config "/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux/openjdk-tests/TKG/../../jvmtest/functional/cmdLineTests/jvmtitests/decompilationtests.xml" \
13:58:17  -explainExcludes -xids all,linux_ppc -xlist "/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux/openjdk-tests/TKG/../../jvmtest/functional/cmdLineTests/jvmtitests/jvmtitests_excludes_16.xml" -nonZeroExitWhenError; \
13:58:17  if [ $? -eq 0 ] ; then echo ""; echo "cmdLineTester_decompilationTests_0""_PASSED"; echo ""; cd /home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux/openjdk-tests/TKG/..; rm -f -r "/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux/openjdk-tests/TKG/../TKG/output_16154637976441/cmdLineTester_decompilationTests_0"; else echo ""; echo "cmdLineTester_decompilationTests_0""_FAILED"; echo ""; fi; } 2>&1 | tee -a "/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux/openjdk-tests/TKG/../TKG/output_16154637976441/TestTargetResult";
13:58:17  Unhandled exception
13:58:17  Type=Illegal instruction vmState=0x00000000
13:58:17  J9Generic_Signal_Number=00000048 Signal_Number=00000004 Error_Value=00000000 Signal_Code=00000001
13:58:17  Handler1=0000FFFF9410A0E0 Handler2=0000FFFF940813E8
13:58:17  R0=00000007FFF28180 R1=0000FFFF6EB81050 R2=000000000000000E R3=0000FFFF90053D80
13:58:17  R4=00000000000ABF60 R5=000000000000A00F R6=0000000000000000 R7=0000000000000001
13:58:17  R8=0000FFFF6EB81050 R9=0000FFFF9001D618 R10=FFFFFFFFFFFFFFFF R11=0000000000000001
13:58:17  R12=0000000000000000 R13=0000FFFF9001D5F0 R14=0000FFFF9001D5F8 R15=0000000000000000
13:58:17  R16=0000FFFF940570D8 R17=0000FFFF94F33ED0 R18=00000007FFF0FA08 R19=0000000000011200
13:58:17  R20=00000000000ABBB0 R21=0000000000000000 R22=00000006800113F8 R23=00000007FFF28180
13:58:17  R24=0000000000034A00 R25=0000000000600000 R26=00000007FFF051F8 R27=00000007FFF04F20
13:58:17  R28=000000000002EE3A R29=0000005B0000006E R30=0000FFFF6EA370C8 R31=0000FFFF94D869C0
13:58:17  PC=0000FFFF6EAB3710 SP=0000FFFF94D869C0 PSTATE=0000000000000000
13:58:17  V0 0000000000000000 (f: 0.000000, d: 0.000000e+00)
13:58:17  V1 945851ba2701453c (f: 654394688.000000, d: -1.155828e-210)
13:58:17  V2 0000ffff945851ba (f: 2488816128.000000, d: 1.390662e-309)
13:58:17  V3 bfe0286bb9ec675a (f: 3119277824.000000, d: -5.049342e-01)
13:58:17  V4 bfd00ea348b88334 (f: 1220051712.000000, d: -2.508934e-01)
13:58:17  V5 3fd5575b0be00b6a (f: 199232368.000000, d: 3.334568e-01)
13:58:17  V6 bfdffffef20a4123 (f: 4060758272.000000, d: -4.999997e-01)
13:58:17  V7 3fe62e42fefa39ef (f: 4277811712.000000, d: 6.931472e-01)
13:58:17  V8 0000000000000000 (f: 0.000000, d: 0.000000e+00)
13:58:17  V9 0000000000000000 (f: 0.000000, d: 0.000000e+00)
13:58:17  V10 0000000000000000 (f: 0.000000, d: 0.000000e+00)
13:58:17  V11 0000000000000000 (f: 0.000000, d: 0.000000e+00)
13:58:17  V12 0000000000000000 (f: 0.000000, d: 0.000000e+00)
13:58:17  V13 0000000000000000 (f: 0.000000, d: 0.000000e+00)
13:58:17  V14 0000000000000000 (f: 0.000000, d: 0.000000e+00)
13:58:17  V15 0000000000000000 (f: 0.000000, d: 0.000000e+00)
13:58:17  V16 bff0000000000000 (f: 0.000000, d: -1.000000e+00)
13:58:17  V17 00000000aaa80000 (f: 2863136768.000000, d: 1.414578e-314)
13:58:17  V18 0000000000000000 (f: 0.000000, d: 0.000000e+00)
13:58:17  V19 0000ffff8f75e418 (f: 2406867968.000000, d: 1.390662e-309)
13:58:17  V20 0000000000000000 (f: 0.000000, d: 0.000000e+00)
13:58:17  V21 0000000000000000 (f: 0.000000, d: 0.000000e+00)
13:58:17  V22 0000000000000000 (f: 0.000000, d: 0.000000e+00)
13:58:17  V23 0000000000000000 (f: 0.000000, d: 0.000000e+00)
13:58:17  V24 0000000000000000 (f: 0.000000, d: 0.000000e+00)
13:58:17  V25 0000000000000000 (f: 0.000000, d: 0.000000e+00)
13:58:17  V26 0000000000000000 (f: 0.000000, d: 0.000000e+00)
13:58:17  V27 0000000000000000 (f: 0.000000, d: 0.000000e+00)
13:58:17  V28 0000000000000000 (f: 0.000000, d: 0.000000e+00)
13:58:17  V29 0000000000000000 (f: 0.000000, d: 0.000000e+00)
13:58:17  V30 0000000000000000 (f: 0.000000, d: 0.000000e+00)
13:58:17  V31 0000000041040000 (f: 1090781184.000000, d: 5.389175e-315)
13:58:17  
13:58:17  Compiled_method=unknown (In JIT code segment 0000FFFF90137C98 but no method found)
13:58:17  Target=2_90_20210311_79 (Linux 4.15.0-34-generic)
13:58:17  CPU=aarch64 (64 logical CPUs) (0x1f699a1000 RAM)
13:58:17  ----------- Stack Backtrace -----------
13:58:17  (0x0000FFFF9407E040 [libj9prt29.so+0x26040])
13:58:17  ---------------------------------------
13:58:17  JVMDUMP039I Processing dump event "gpf", detail "" at 2021/03/11 13:58:16 - please wait.
13:58:17  JVMDUMP032I JVM requested System dump using '/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux/openjdk-tests/TKG/output_16154637976441/cmdLineTester_decompilationTests_0/core.20210311.135816.37296.0001.dmp' in response to an event
13:58:17  JVMDUMP010I System dump written to /home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux/openjdk-tests/TKG/output_16154637976441/cmdLineTester_decompilationTests_0/core.20210311.135816.37296.0001.dmp
13:58:17  JVMDUMP032I JVM requested Java dump using '/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux/openjdk-tests/TKG/output_16154637976441/cmdLineTester_decompilationTests_0/javacore.20210311.135816.37296.0002.txt' in response to an event
13:58:18  JVMDUMP010I Java dump written to /home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux/openjdk-tests/TKG/output_16154637976441/cmdLineTester_decompilationTests_0/javacore.20210311.135816.37296.0002.txt
13:58:18  JVMDUMP032I JVM requested Snap dump using '/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux/openjdk-tests/TKG/output_16154637976441/cmdLineTester_decompilationTests_0/Snap.20210311.135816.37296.0003.trc' in response to an event
13:58:18  JVMDUMP010I Snap dump written to /home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux/openjdk-tests/TKG/output_16154637976441/cmdLineTester_decompilationTests_0/Snap.20210311.135816.37296.0003.trc
13:58:18  JVMDUMP032I JVM requested JIT dump using '/home/jenkins/workspace/Test_openjdk16_j9_sanity.functional_aarch64_linux/openjdk-tests/TKG/output_16154637976441/cmdLineTester_decompilationTests_0/jitdump.20210311.135816.37296.0004.dmp' in response to an event
13:58:18  JVMDUMP013I Processed dump event "gpf", detail "".
13:58:18  
13:58:18  cmdLineTester_decompilationTests_0_FAILED
```

**Comment by pshipton** [comment]
@knn-k fyi

**Comment by knn-k** [comment]
Grinder run failed in https://ci.adoptopenjdk.net/job/Grinder/7828/tapResults/

**Comment by knn-k** [comment]
It looks like the same problem as that in https://github.com/eclipse/openj9/issues/11851#issuecomment-784711295
The code generated by the JIT is so huge that the target of conditional branch instruction is out of its range.

There is a workaround for huge methods in OMR master branch: https://github.com/eclipse/omr/pull/5819
Should I apply the workaround for v0.25.0, or is it too late?

**Comment by pshipton** [comment]
It's too late for 0.25.0 as it has already been released. The fix is in the 0.26.0 release branch already, which will GA around April 23.

**Comment by pshipton** [comment]
fyi https://github.com/AdoptOpenJDK/TSC/issues/211
